### PR TITLE
feat: migrate to connector.send_tx

### PR DESCRIPTION
**Pull request by justprosh**

### CODE REVIEW

File: src/services/chain-connector/modules/chain_connector/src/jsonrpc/register_worker.rs [hunk 1]
```diff
@@ -1,129 +1,65 @@
...
-        let key = parse_wallet_key(&chain.wallet_key)?;
-        let input = encode_call(unit_id.clone(), worker_id)?;
-        let nonce = load_nonce(key.to_address(), endpoint)?;
-        let gas_price = get_gas_price(endpoint)?;
-        let tx = make_tx(input, key, gas, nonce, gas_price, deal_addr, network_id)?;
-        log::debug!(
-            "wallet {}; wallet address {}; unit_id {}; worker_id {worker_id}; nonce {nonce}; gas_price {gas_price}", 
-            chain.wallet_key, key.to_address(), hex::encode(unit_id)
-        );
-        log::debug!("tx {tx}");
```
Comment (folex) [inline src/services/chain-connector/modules/chain_connector/src/jsonrpc/register_worker.rs:87]: keep these please

File: src/tests/decider-distro-tests-rs/tests/utils/default.rs [hunk 2]
```diff
@@ -13,6 +13,10 @@ pub const DEAL_IDS: &[&'static str] = &[
 
 pub const IPFS_MULTIADDR: &str = "/ip4/127.0.0.1/tcp/5001";
 
+pub const WALLET_KEY: &str = "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f";
```
Comment (folex) [inline src/tests/decider-distro-tests-rs/tests/utils/default.rs:16]:
```suggestion
/// Wallet Private Key to pass to Nox in tests
pub const WALLET_KEY: &str = "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f";
```

File: src/tests/decider-distro-tests-rs/tests/utils/setup.rs [hunk 3]
```diff
@@ -37,6 +41,17 @@ pub async fn setup_swarm(distro: PackageDistro, peers: usize) -> Vec<CreatedSwar
         // to make worker spell oneshot
         config.decider.worker_period_sec = 0;
         cfg.override_system_services_config = Some(config);
+
+        let chain_info = distro.spells[0].kv["chain"].as_object().unwrap();
+        println!("chain_info: {:?}", chain_info);
```
Comment (folex) [inline src/tests/decider-distro-tests-rs/tests/utils/setup.rs:46]: ?

**Review by folex — APPROVED** [review]

File: src/tests/decider-distro-tests-rs/tests/utils/setup.rs (hunk 3 above)
Comment (justprosh) [reply src/tests/decider-distro-tests-rs/tests/utils/setup.rs:46]: used in debug

**Pull request merged**